lib/util.ts

```diff
@@ -12,20 +10,28 @@ const SLSH = { '0': 0, 't': 9, 'n': 10, 'v': 11, 'f': 12, 'r': 13 };
  * @param {string} str
  * @return {string}
  */
-exports.strToChars = (str) => {
+export const strToChars = (str: string) => {
   /* jshint maxlen: false */
   const charsRegex = /(\[\\b\])|(\\)?\\(?:u([A-F0-9]{4})|x([A-F0-9]{2})|(0?[0-7]{2})|c([@A-Z[\\\]^?])|([0tnvfr]))/g;
   return str.replace(charsRegex, (s, b, lbs, a16, b16, c8, dctrl, eslsh) => {
     if (lbs) {
       return s;
     }
 
-    let code = b ? 8 :
+    let code: number | undefined = b ? 8 :
       a16   ? parseInt(a16, 16) :
       b16   ? parseInt(b16, 16) :
       c8    ? parseInt(c8,   8) :
       dctrl ? CTRL.indexOf(dctrl) :
-      SLSH[eslsh];
+      eslsh == '0' ? 0 :
+      eslsh == 't' ? 9 :
+      eslsh == 'n' ? 10 :
+      eslsh == 'v' ? 11 :
+      eslsh == 'f' ? 12 :
+      eslsh == 'r' ? 13 : undefined;
+
+    if (!code)
+      throw new Error(`Code is undefined`)
```

Comment:
add a semicolon. also prefer that multiline `if` statements use curly brackets `{}`. i'll try to add an `eslint` checker after all this.